processor/signoztailsampler/config_test.go

```diff
@@ -42,136 +27,34 @@ func TestLoadConfig(t *testing.T) {
 			NumTraces:               100,
 			ExpectedNewTracesPerSec: 10,
 			PolicyCfgs: []PolicyCfg{
 				{
-					sharedPolicyCfg: sharedPolicyCfg{
-						Name: "test-policy-1",
-						Type: AlwaysSample,
-					},
-				},
-				{
-					sharedPolicyCfg: sharedPolicyCfg{
-						Name:       "test-policy-2",
-						Type:       Latency,
-						LatencyCfg: LatencyCfg{ThresholdMs: 5000},
-					},
-				},
-				{
-					sharedPolicyCfg: sharedPolicyCfg{
-						Name:                "test-policy-3",
-						Type:                NumericAttribute,
-						NumericAttributeCfg: NumericAttributeCfg{Key: "key1", MinValue: 50, MaxValue: 100},
-					},
-				},
-				{
-					sharedPolicyCfg: sharedPolicyCfg{
-						Name:             "test-policy-4",
-						Type:             Probabilistic,
-						ProbabilisticCfg: ProbabilisticCfg{HashSalt: "custom-salt", SamplingPercentage: 0.1},
-					},
-				},
-				{
-					sharedPolicyCfg: sharedPolicyCfg{
-						Name:          "test-policy-5",
-						Type:          StatusCode,
-						StatusCodeCfg: StatusCodeCfg{StatusCodes: []string{"ERROR", "UNSET"}},
-					},
-				},
-				{
-					sharedPolicyCfg: sharedPolicyCfg{
-						Name:               "test-policy-6",
-						Type:               StringAttribute,
-						StringAttributeCfg: StringAttributeCfg{Key: "key2", Values: []string{"value1", "value2"}},
-					},
-				},
-				{
-					sharedPolicyCfg: sharedPolicyCfg{
-						Name:            "test-policy-7",
-						Type:            RateLimiting,
-						RateLimitingCfg: RateLimitingCfg{SpansPerSecond: 35},
-					},
-				},
-				{
-					sharedPolicyCfg: sharedPolicyCfg{
-						Name:         "test-policy-8",
-						Type:         SpanCount,
-						SpanCountCfg: SpanCountCfg{MinSpans: 2},
-					},
-				},
-				{
-					sharedPolicyCfg: sharedPolicyCfg{
-						Name:          "test-policy-9",
-						Type:          TraceState,
-						TraceStateCfg: TraceStateCfg{Key: "key3", Values: []string{"value1", "value2"}},
+					Name:     "test-policy-1",
+					Type:     PolicyGroup,
+					Priority: 1,
+					Root:     true,
+					ProbabilisticCfg: ProbabilisticCfg{
+						SamplingPercentage: 100,
 					},
-				},
-				{
-					sharedPolicyCfg: sharedPolicyCfg{
-						Name: "and-policy-1",
-						Type: And,
-					},
-					AndCfg: AndCfg{
-						SubPolicyCfg: []AndSubPolicyCfg{
-							{
-								sharedPolicyCfg: sharedPolicyCfg{
-									Name:                "test-and-policy-1",
-									Type:                NumericAttribute,
-									NumericAttributeCfg: NumericAttributeCfg{Key: "key1", MinValue: 50, MaxValue: 100},
-								},
-							},
-							{
-								sharedPolicyCfg: sharedPolicyCfg{
-									Name:               "test-and-policy-2",
-									Type:               StringAttribute,
-									StringAttributeCfg: StringAttributeCfg{Key: "key2", Values: []string{"value1", "value2"}},
-								},
-							},
+					PolicyFilterCfg: PolicyFilterCfg{
+						StringAttributeCfgs: []StringAttributeCfg{
+							{Key: "source", Values: []string{"security"}},
 						},
```

Comment:
yes. If sub-policy has a sampling rate different then it will take precedence. If none of the sub-policies apply then the root policy sampling rate will be applied.